### PR TITLE
Upgrade pulumi/pulumi

### PR DESCRIPTION
**Pull request by blampe**
Bumps pu/pu across the entire repo, including a few examples that were still using very old versions.

### CODE REVIEW

**Review by guineveresaenger — CHANGES_REQUESTED** [review]
While this probably works, it is not recommended to have the Pulumi version overtake the [pulumi version used in the bridge](https://github.com/pulumi/pulumi-terraform-bridge/blob/master/x/muxer/go.mod#L75).

I recommend keeping the pu/pu version currently in go.mod and updating the examples files to that.

Also this is somewhere between a missing feature in `upgrade-provider` and the setup of the Go tests in this repo:
- `upgrade-provider` does not detect _all_ go.mod files in the examples
- this is because most providers expect a single go.mod file in `examples/, not one per example.

**Comment by blampe** [comment]
> While this probably works, it is not recommended to have the Pulumi version overtake the [pulumi version used in the bridge](https://github.com/pulumi/pulumi-terraform-bridge/blob/master/x/muxer/go.mod#L75).
> 
> I recommend keeping the pu/pu version currently in go.mod and updating the examples files to that.

Good catch, updated.
 
> Also this is somewhere between a missing feature in `upgrade-provider` and the setup of the Go tests in this repo:
> 
> * `upgrade-provider` does not detect _all_ go.mod files in the examples
> * this is because most providers expect a single go.mod file in `examples/, not one per example.

Yes, we want behavior analogous to https://github.com/pulumi/ci-mgmt/pull/778 for bridged providers.

`for MODFILE in $(find . -name go.mod); do pushd $(dirname $MODFILE); go get github.com/pulumi/pulumi-terraform-bridge/v3 ; go mod tidy; popd; done`

I think this is essentially https://github.com/pulumi/upgrade-provider/issues/142

File: examples/aws-container-registry/go/go.mod [hunk 1]
```diff
@@ -5,7 +5,7 @@ go 1.21
 require (
 	github.com/pulumi/pulumi-aws/sdk/v5 v5.30.0
 	github.com/pulumi/pulumi-docker/sdk/v4 v4.0.0-alpha.4
-	github.com/pulumi/pulumi/sdk/v3 v3.101.1
+	github.com/pulumi/pulumi/sdk/v3 v3.99.0
```
Comment (t0yv0) [inline examples/aws-container-registry/go/go.mod:8]: Downgrade?

**Comment by t0yv0** [comment]
Yeah I'm in the  process of finishing up the bridge test on 101 https://github.com/pulumi/pulumi-terraform-bridge/pull/1624 - as far as I can tell the p/p codegen version passes all tests, so if you want to make an exception and move forward that's fine with me, but normally we get the p/p dependency bumped automatically through bridge updates which should be coming shortly once I'm done validating the above and clearing any blockers.

File: examples/aws-container-registry/go/go.mod (hunk 1 above)
Comment (blampe) [reply examples/aws-container-registry/go/go.mod:8]: I think you were looking at the changes between revisions, this is upgrading from 3.55 to 3.99.

**Comment by blampe** [comment]
> normally we get the p/p dependency bumped automatically through bridge updates

@t0yv0 the problem is the updates aren't comprehensive, see my other comment here https://github.com/pulumi/pulumi-docker/pull/945#issuecomment-1897524081

**Comment by t0yv0** [comment]
Right. I thought you correctly identified the root cause on that we need to hack update-provider a little further.

**Review by t0yv0 — APPROVED** [review]

File: examples/aws-container-registry/go/go.mod (hunk 1 above)
Comment (t0yv0) [reply examples/aws-container-registry/go/go.mod:8]: LGTM then thank you!

**Review by guineveresaenger — APPROVED** [review]

**Pull request merged**